core/bin/zksync_tee_prover/src/api_client.rs

```diff
@@ -0,0 +1,146 @@
+use reqwest::Client;
+use secp256k1::{ecdsa::Signature, PublicKey, Secp256k1, SecretKey};
+use serde::{de::DeserializeOwned, Serialize};
+use url::Url;
+use zksync_basic_types::H256;
+use zksync_prover_interface::{
+    api::{
+        RegisterTeeAttestationRequest, RegisterTeeAttestationResponse, SubmitProofResponse,
+        SubmitTeeProofRequest, TeeProofGenerationDataRequest, TeeProofGenerationDataResponse,
+    },
+    inputs::TeeVerifierInput,
+    outputs::L1BatchTeeProofForL1,
+};
+use zksync_types::{tee_types::TeeType, L1BatchNumber};
+
+/// Configuration for the [`zksync_proof_data_handler::run_server`] API client that serves the TEE
+/// prover by:
+/// 1. Providing proof generation data.
+/// 2. Submitting proofs.
+/// 3. Submitting attestation quotes.
+#[derive(Debug)]
+pub(crate) struct ApiClient {
+    /// Base URL for the [`zksync_proof_data_handler`] API. Optionally, include the port number,
+    /// e.g., `https://127.0.0.1:3320`.
+    api_base_url: Url,
+    /// HTTP client used to send requests to the API.
+    http_client: Client,
+}
+
+impl ApiClient {
+    pub fn new(api_base_url: Url, http_client: Client) -> Self {
+        ApiClient {
+            api_base_url,
+            http_client,
+        }
+    }
+
+    async fn send_http_request<Req, Resp>(
+        &self,
+        request: Req,
+        endpoint: Url,
```

Comment:
Two things here:
1. The ordering of arguments seems odd, naturally you'd pass endpoint first and request after.
2. You may take `AsRef<str>` as an argument here and join inside.

This will make the interface nicer, e.g. `self.post("/tee/register_attestation", request)`.

Reply:
Copy-paste from here:
https://github.com/matter-labs/zksync-era/blob/fe03d0e254a98fea60ecb7485a7de9e7fdecaee1/prover/prover_fri_gateway/src/api_data_fetcher.rs#L27-L46

Good suggestion anyway. I will fix it. 👍🏼

Reply:
Copy-paste from here:
https://github.com/matter-labs/zksync-era/blob/fe03d0e254a98fea60ecb7485a7de9e7fdecaee1/prover/prover_fri_gateway/src/api_data_fetcher.rs#L27-L46

Good suggestion anyway. I will fix it. 👍🏼